scm-webapp/src/test/java/sonia/scm/api/v2/resources/ConfigResourceTest.java

```diff
@@ -77,100 +75,141 @@
-      ThreadContext.remove();
-  }
-
-  @Before
-  public void prepareEnvironment() {
-    initMocks(this);
-
+  @BeforeEach
+  void prepareEnvironment() {
     ConfigResource configResource = new ConfigResource(dtoToConfigMapper, configToDtoMapper, createConfiguration(), namespaceStrategyValidator);
-    configResource.setStore(config -> {});
+    configResource.setStore(config -> {
+    });
 
     dispatcher.addSingletonResource(configResource);
+
+    ThreadContext.bind(subject);
+  }
+
+  @AfterEach
+  void tearDown() {
+    ThreadContext.unbindSubject();
   }
 
   @Test
-  @SubjectAware(username = "readOnly")
-  public void shouldGetGlobalConfig() throws URISyntaxException, UnsupportedEncodingException {
+  void shouldGetGlobalConfig() throws URISyntaxException, UnsupportedEncodingException {
+    doNothing().when(subject).checkPermission("configuration:read:global");
+
     MockHttpRequest request = MockHttpRequest.get("/" + ConfigResource.CONFIG_PATH_V2);
     MockHttpResponse response = new MockHttpResponse();
     dispatcher.invoke(request, response);
-    assertEquals(HttpServletResponse.SC_OK, response.getStatus());
-    assertTrue(response.getContentAsString().contains("\"proxyPassword\":\"heartOfGold\""));
-    assertTrue(response.getContentAsString().contains("\"self\":{\"href\":\"/v2/config"));
-    assertFalse("Update link present", response.getContentAsString().contains("\"update\":{\"href\":\"/v2/config"));
+
+    assertThat(response.getStatus()).isEqualTo(HttpServletResponse.SC_OK);
+    assertThat(response.getContentAsString()).contains("\"proxyPassword\":\"heartOfGold\"");
+    assertThat(response.getContentAsString()).contains("\"self\":{\"href\":\"/v2/config");
+    assertThat(response.getContentAsString()).doesNotContain("\"update\":{\"href\":\"/v2/config");
   }
 
   @Test
-  @SubjectAware(username = "writeOnly")
-  public void shouldNotGetConfigWhenNotAuthorized() throws URISyntaxException, UnsupportedEncodingException {
+  void shouldNotGetConfigWhenNotAuthorized() throws URISyntaxException {
+    doThrow(AuthorizationException.class).when(subject).checkPermission("configuration:read:global");
+
     MockHttpRequest request = MockHttpRequest.get("/" + ConfigResource.CONFIG_PATH_V2);
     MockHttpResponse response = new MockHttpResponse();
 
     dispatcher.invoke(request, response);
 
-    assertEquals("Subject does not have permission [configuration:read:global]", response.getContentAsString());
-    assertEquals(HttpServletResponse.SC_FORBIDDEN, response.getStatus());
+    assertThat(response.getStatus()).isEqualTo(HttpServletResponse.SC_FORBIDDEN);
   }
 
   @Test
-  @SubjectAware(username = "readWrite")
-  public void shouldUpdateConfig() throws URISyntaxException, IOException {
-    MockHttpRequest request = post("sonia/scm/api/v2/config-test-update.json");
+  void shouldUpdateConfig() throws URISyntaxException, IOException {
+    doNothing().when(subject).checkPermission("configuration:write:global");
```

Comment:
There is now a nicer way to write tests for Shiro using JUnit 5.

https://github.com/sdorra/junit-shiro-extension

Reply:
Oh, i missed this. Thanks. Will try it out.